src/VectorFiber.jl

```diff
@@ -0,0 +1,279 @@
+
+"""
+    VectorSpaceFiberType{TVS<:VectorSpaceType}
+
+`FiberType` of a [`FiberBundle`](@ref) corresponding to [`VectorSpaceType`](@ref)
+`TVS`.
+"""
+struct VectorSpaceFiberType{TVS<:VectorSpaceType} <: FiberType
+    fiber::TVS
+end
+
+function Base.show(io::IO, vsf::VectorSpaceFiberType)
+    return print(io, "VectorSpaceFiberType($(vsf.fiber))")
+end
+
+const TangentFiberType = VectorSpaceFiberType{TangentSpaceType}
+
+const TangentFiber = VectorSpaceFiberType{TangentSpaceType}(TangentSpace)
+
+"""
+    VectorBundleFibers{TVS,TM}
+
+Alias for [`BundleFibers`](@ref) when the fiber is a vector space.
+"""
+const VectorBundleFibers{TVS,TM} = BundleFibers{
+    VectorSpaceFiberType{TVS},
+    TM,
+} where {TVS<:VectorSpaceType,TM<:AbstractManifold}
+
+function VectorBundleFibers(fiber::VectorSpaceType, M::AbstractManifold)
+    return BundleFibers(VectorSpaceFiberType(fiber), M)
+end
+function VectorBundleFibers(fiber::VectorSpaceFiberType, M::AbstractManifold)
+    return BundleFibers(fiber, M)
+end
+
+const TangentBundleFibers{M} = BundleFibers{TangentFiberType,M} where {M<:AbstractManifold}
+
+TangentBundleFibers(M::AbstractManifold) = BundleFibers(TangentFiber, M)
+
+"""
+    VectorSpaceAtPoint{𝔽,M,TFiber}
+
+Alias for [`FiberAtPoint`](@ref) when the fiber is a vector space.
+"""
+const VectorSpaceAtPoint{𝔽,M,TSpaceType} = FiberAtPoint{
+    𝔽,
+    VectorBundleFibers{TSpaceType,M},
+} where {𝔽,M<:AbstractManifold{𝔽},TSpaceType<:VectorSpaceType}
+
+function VectorSpaceAtPoint(M::AbstractManifold, fiber::VectorSpaceFiberType, p)
+    return FiberAtPoint(BundleFibers(fiber, M), p)
+end
+
+"""
+    TangentSpaceAtPoint{𝔽,M}
+
+Alias for [`VectorSpaceAtPoint`](@ref) for the tangent space at a point.
+"""
+const TangentSpaceAtPoint{𝔽,M} =
+    VectorSpaceAtPoint{𝔽,M,TangentSpaceType} where {𝔽,M<:AbstractManifold{𝔽}}
+
+"""
+    TangentSpaceAtPoint(M::AbstractManifold, p)
+
+Return an object of type [`VectorSpaceAtPoint`](@ref) representing tangent
+space at `p` on the [`AbstractManifold`](@ref) `M`.
+"""
+TangentSpaceAtPoint(M::AbstractManifold, p) = VectorSpaceAtPoint(M, TangentFiber, p)
+
+"""
+    TangentSpace(M::AbstractManifold, p)
+
+Return a [`TangentSpaceAtPoint`] representing tangent
+space at `p` on the [`AbstractManifold`](@ref) `M`.
+"""
+TangentSpace(M::AbstractManifold, p) = TangentSpaceAtPoint(M, p)
```

Comment:
I currently find the use of `TangentSpace` at least misleading.
It is used here as a short hand for `TangentSpaceAtPoint` (why were we so precise with that name anyways?) but it is also used as

```
const TangentSpace = TangentSpaceType()
```

And I am not yet sure what `TangentSpaceType is necessary for. Could we remove this naming double use somehow?

Reply:
Also I carefully read the doc string of `VectorSpaceType` (very beginning of bases.jl) and I am not yet sure I understand why we need that and what it is necessary for (even less with the const in the last comment).

Reply:
`TangentSpaceType` is used to distinguish tangent bundle from other vector bundles, and tangent spaces at point from other spaces (at point). We could remove `TangentSpace` constant if you think that would be more clear but `TangentSpaceType` is necessary (though it could be renamed if needed). This particular method is definitely not needed.

Renaming `TangentSpaceAtPoint` to `TangentSpace` is possible but that would require also renaming other `AtPoint` types. I'm not sure if new names would be more or less confusing.

Reply:
Do you think it would be better to rename `TangentSpaceType` to `TangentSpace` and replace `TangentSpace` with ``TangentSpace()`?

Reply:
Hm, I think I would maybe do the opposite, rename the `AtPoint` things to without that, so we have a (user-facing) `TangentSpace` type that represents – a tangent space.

`TangentSpaceType` is something more technical / design decision, not that public, so I think there it is fine to have the more complicated name?
And the reason why `TangentSpace=TangentSpaceType()` is what we currently have is also something I do not understand especially if that later appears as a type but is actually an instance?

Reply:
I see, I will try that.